src/client/coll/mod.rs

```diff
@@ -0,0 +1,301 @@
+pub mod options;
+pub mod results;
+
+use bson;
+use bson::Bson;
+
+use client::db::Database;
+use client::common::{ReadPreference, WriteConcern};
+use client::coll::options::*;
+use client::coll::results::*;
+
+use client::wire_protocol::flags::OpQueryFlags;
+use client::wire_protocol::operations::Message;
+
+/// Interfaces with a MongoDB collection.
+pub struct Collection<'a> {
+    db: &'a Database<'a>,
+    pub namespace: String,
+    read_preference: ReadPreference,
+    write_concern: WriteConcern,
+}
+
+impl<'a> Collection<'a> {
+    /// Creates a collection representation with optional read and write controls.
+    ///
+    /// If `create` is specified, the collection will be explicitly created in the database.
+    pub fn new(db: &'a Database<'a>, name: &str, create: bool,
+               read_preference: Option<ReadPreference>, write_concern: Option<WriteConcern>) -> Collection<'a> {
+
+        let rp = match read_preference {
+            Some(rp) => rp,
+            None => db.read_preference.to_owned(),
+        };
+
+        let wc = match write_concern {
+            Some(wc) => wc,
+            None => db.write_concern.to_owned(),
+        };
+
+        Collection {
+            db: db,
+            namespace: format!("{}.{}", db.name, name),
+            read_preference: rp,
+            write_concern: wc,
+        }
+    }
+
+    /// Returns a unique operational request id.
+    pub fn get_req_id(&self) -> i32 {
+        self.db.client.get_req_id()
+    }
+
+    /// Extracts the collection name from the namespace.
+    pub fn name(&self) -> String {
+        match self.namespace.find(".") {
+            Some(idx) => self.namespace[idx+1..].to_owned(),
```

Comment:
careful with this. This is a byte index of these strings, are you guaranteed that this is actually correct? Can namespaces have non-ascii characters?